huaweicloud/services/cdn/data_source_huaweicloud_cdn_top_url_statistics.go

```diff
@@ -0,0 +1,200 @@
+
+			// Attributes
+			"top_url_statistics": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        topUrlStatisticsSchema(),
+				Description: `The list of TOP100 URL statistics that matched filter parameters.`,
+			},
+		},
+	}
+}
+
+func topUrlStatisticsSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"url": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The URL name.`,
+			},
+			"value": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `The value corresponding to the query type.`,
+			},
+			"start_time": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The start time of the query range, in RFC3339 format.`,
+			},
+			"end_time": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The end time of the query range, in RFC3339 format.`,
+			},
+			"stat_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The statistical type of the query.`,
+			},
+		},
+	}
+}
+
+func buildTopUrlStatisticsQueryParams(d *schema.ResourceData) string {
+	res := "?"
+
+	// Convert RFC3339 format str to timestamp
+	if v, ok := d.GetOk("start_time"); ok {
+		res = fmt.Sprintf("%s&start_time=%v", res, utils.ConvertTimeStrToNanoTimestamp(v.(string)))
+	}
+	// Convert RFC3339 format str to timestamp
+	if v, ok := d.GetOk("end_time"); ok {
+		res = fmt.Sprintf("%s&end_time=%v", res, utils.ConvertTimeStrToNanoTimestamp(v.(string)))
+	}
+	if v, ok := d.GetOk("domain_name"); ok {
+		res = fmt.Sprintf("%s&domain_name=%v", res, v)
+	}
+	if v, ok := d.GetOk("stat_type"); ok {
+		res = fmt.Sprintf("%s&stat_type=%v", res, v)
+	}
+	if v, ok := d.GetOk("service_area"); ok {
+		res = fmt.Sprintf("%s&service_area=%v", res, v)
+	}
+	if v, ok := d.GetOk("enterprise_project_id"); ok {
+		res = fmt.Sprintf("%s&enterprise_project_id=%v", res, v)
+	}
+
+	return res
+}
+
+func flattenTopUrlStatistics(items []interface{}) []map[string]interface{} {
+	if len(items) < 1 {
+		return nil
+	}
+
+	result := make([]map[string]interface{}, 0, len(items))
+	for _, item := range items {
+		result = append(result, map[string]interface{}{
+			"url":        utils.PathSearch("url", item, nil),
+			"value":      utils.PathSearch("value", item, nil),
+			"start_time": utils.FormatTimeStampRFC3339(int64(utils.PathSearch("start_time", item, float64(0)).(float64))/1000, true),
+			"end_time":   utils.FormatTimeStampRFC3339(int64(utils.PathSearch("end_time", item, float64(0)).(float64))/1000, true),
+			"stat_type":  utils.PathSearch("stat_type", item, nil),
+		})
+	}
+
+	return result
+}
+
+func resourceTopUrlStatisticsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v1.0/cdn/statistics/top-url"
+	)
+
+	client, err := cfg.NewServiceClient("cdn", region)
```

Comment:
```suggestion
	client, err := cfg.NewServiceClient("cdn", "")
```
This is a global service, it does not have the region parameter

Reply:
done